### ISSUE TITLE
[SERVICE]: GITLAB Self Hosted

### ISSUE THREAD

**Issue by compgeniuses** [coollabsio/coolify#2635]
### Description

I would Like a Full implementation of GITLAB as a Full Featured Service Within Coolify

Should Also include corresponding runners as well.

### Minimal Reproduction (if possible, example repository)

https://docs.gitlab.com/ee/install/docker.html

### Version

4

**Comment by GunniBusch** [comment]
/attempt #2635

I put a rough compose file together. needs a bit of refining but I will open a pr with more details:

```yaml
services:
  gitlab:
    image: 'gitlab/gitlab-ce:latest'
    container_name: gitlab
    restart: always
    hostname: '${SERVICE_URL_GITLAB}'
    environment:
      - SERVICE_FQDN_GITLAB_80
      - EXTERNAL_URL=$SERVICE_FQDN_GITLAB
      - GITLAB_HOST=$SERVICE_FQDN_GITLAB
      - GITLAB_OMNIBUS_CONFIG="external_url=${SERVICE_FQDN_GITLAB}; gitlab_rails['gitlab_shell_ssh_port'] = 2222"
    ports:
      - '2222:22'
    volumes:
      - 'gitlab-config:/etc/gitlab'
      - 'gitlab-logs:/var/log/gitlab'
      - 'gitlab-data:/var/opt/gitlab'
    shm_size: 256m
```

**Comment by compgeniuses** [comment]
Wonderful News.

**Comment by andrasbacsai** [comment]
@compgeniuses when this issue is closed by #2941, please accept the bounty.
